### DISCUSSION TITLE
Create a parametrized fixture

### DISCUSSION THREAD

**Discussion by Gadzillion** [pytest-dev/pytest#8062] [Q&A] [answered]
I want to create a parametrized fixture that gets args (for example two lists) from a test, performs some actions on it, return one list (based on two given), and uses one value at a time from the list (code below doesn't work, but it shows my idea):

```
LIST_1 = [1, 3]
LIST_2 = [1, 2]

@pytest.fixture(scope="function", params=[request])
def tester(request):
    some_actions ...
    [I for I in request.param[0] if I not in request.param[1]]
    return ... (return one value at a time (like request.param). for example (1, 1) , then (3, 2))

@pytest.mark.parametrize('tester', [(LSIT_1, LIST_2)], indirect=True)
def test_button(tester):
    ...
```

**Comment by The-Compiler** [comment, marked as answer]
I don't think you can do this directly from a fixture. You could either use parametrize and some helper function:

```python
import pytest

LIST_1 = [1, 3]
LIST_2 = [1, 2]

def gen_params(elems1, elems2):
    return list(zip(elems1, elems2))

@pytest.mark.parametrize('tester', gen_params(LIST_1, LIST_2))
def test_button(tester):
    assert False, tester
```

or use [a custom mark](https://docs.pytest.org/en/stable/example/markers.html#mark-examples) and the [`pytest_generate_tests` hook](https://docs.pytest.org/en/latest/reference.html#pytest.hookspec.pytest_generate_tests) with [`Metafunc.parametrize`](https://docs.pytest.org/en/latest/reference.html#metafunc) to implement custom parametrization.

If you can say more about what your actual use-case is, it might be possible to suggest a simpler/better alternative as well.

**Reply by Gadzillion** [reply]
thx u)
